application.go

```diff
@@ -36,23 +37,23 @@ type Applications struct {
 // Application is the definition for an application in marathon
 type Application struct {
 	ID                    string              `json:"id,omitempty"`
-	Cmd                   string              `json:"cmd,omitempty"`
-	Args                  []string            `json:"args"`
-	Constraints           [][]string          `json:"constraints"`
+	Cmd                   *string             `json:"cmd,omitempty"`
+	Args                  *[]string           `json:"args,omitempty"`
```

Comment:
According to the documentation, either `Cmd` or `Args` must be provided. I haven't tested it but would assume that if you have `Cmd` set on an application and pass `Args` in a `PUT` request, Marathon will use `Args` now (and vice versa).

Have you possibly tested this?